File: dart_sample/lib/demos/material/dialog_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart' as cupertino;
import 'package:material_ui/material_ui.dart';

class DialogDemoPage extends StatefulWidget {
  const DialogDemoPage({super.key});

  @override
  State<DialogDemoPage> createState() => _DialogDemoPageState();
}

class _DialogDemoPageState extends State<DialogDemoPage> {
  bool _scrollable = false;
  bool _barrierDismissible = true;
  bool _useThemeOverrides = false;
  bool _appleAdaptive = false;
  String _lastResult = 'none';

  @override
  Widget build(BuildContext context) {
    final DialogThemeData dialogTheme = _useThemeOverrides
        ? DialogThemeData(
            backgroundColor: Colors.teal.shade50,
            iconColor: Colors.teal.shade800,
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(14),
            ),
            titleTextStyle: Theme.of(
              context,
            ).textTheme.headlineSmall?.copyWith(color: Colors.teal.shade900),
            barrierColor: const Color(0x99004D40),
          )
        : const DialogThemeData();
    return Theme(
      data: Theme.of(context).copyWith(
        dialogTheme: dialogTheme,
        platform: _appleAdaptive ? TargetPlatform.iOS : null,
      ),
      child: Builder(builder: _buildContent),
    );
  }

  Widget _buildContent(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 14,
      children: <Widget>[
        const Text('Dialog family', style: TextStyle(fontSize: 20)),
        const Text(
          'Dialog, AlertDialog, SimpleDialog, typed results, intrinsic width, actions overflow, and scrollable choices.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Wrap(
          spacing: 8,
          runSpacing: 8,
          children: <Widget>[
            TextButton(
              onPressed: () => setState(() => _scrollable = !_scrollable),
              child: Text(_scrollable ? 'Scrollable' : 'Static'),
            ),
            TextButton(
              onPressed: () =>
                  setState(() => _barrierDismissible = !_barrierDismissible),
              child: Text(
                _barrierDismissible ? 'Barrier closes' : 'Barrier locked',
              ),
            ),
            TextButton(
              onPressed: () =>
                  setState(() => _useThemeOverrides = !_useThemeOverrides),
              child: Text(_useThemeOverrides ? 'Theme on' : 'Theme off'),
            ),
            TextButton(
              onPressed: () =>
                  setState(() => _appleAdaptive = !_appleAdaptive),
              child: Text(_appleAdaptive ? 'Apple platform' : 'Host platform'),
            ),
          ],
        ),
        Wrap(
          spacing: 8,
          runSpacing: 8,
          children: <Widget>[
            ElevatedButton(
              onPressed: () => _showAlert(context),
              child: const Text('SHOW ALERT'),
            ),
            OutlinedButton(
              onPressed: () => _showPlainDialog(context),
              child: const Text('SHOW DIALOG'),
            ),
            FilledButton(
              onPressed: () => _showSimpleDialog(context),
              child: const Text('SHOW SIMPLE'),
            ),
            TextButton(
              onPressed: () => _showAdaptive(context),
              child: const Text('SHOW ADAPTIVE'),
            ),
          ],
        ),
        Text('Last result: $_lastResult', style: const TextStyle(fontSize: 13)),
      ],
    );
  }

  Future<void> _showAlert(BuildContext context) async {
    final String? result = await showDialog<String>(
      context: context,
      barrierDismissible: _barrierDismissible,
      builder: (BuildContext routeContext) => AlertDialog(
        icon: const Icon(Icons.info_outline),
        title: const Text('Delete draft?'),
        content: _scrollable
            ? const Column(
                children: <Widget>[
                  Text(
                    'This dialog keeps actions visible while the message scrolls.',
                  ),
                  SizedBox(height: 180),
                  Text('End of the scrollable content.'),
                ],
              )
            : const Text('The draft can be restored later from history.'),
        scrollable: _scrollable,
        actions: <Widget>[
          TextButton(
            onPressed: () => Navigator.pop(routeContext, 'cancel'),
            child: const Text('CANCEL'),
          ),
          TextButton(
            onPressed: () => Navigator.pop(routeContext, 'delete'),
            child: const Text('DELETE'),
          ),
        ],
      ),
    );
    if (mounted) setState(() => _lastResult = result ?? 'dismissed');
  }

  Future<void> _showPlainDialog(BuildContext context) async {
    final String? result = await showDialog<String>(
      context: context,
      barrierDismissible: _barrierDismissible,
      builder: (BuildContext routeContext) => Dialog(
        child: Padding(
          padding: const EdgeInsets.all(24),
          child: Column(
            mainAxisSize: MainAxisSize.min,
            spacing: 16,
            children: <Widget>[
              const Text('Base Dialog', style: TextStyle(fontSize: 20)),
              const Text(
                'This uses the same themed Material surface and route barrier.',
              ),
              TextButton(
                onPressed: () => Navigator.pop(routeContext, 'closed'),
                child: const Text('CLOSE'),
              ),
            ],
          ),
        ),
      ),
    );
    if (mounted) setState(() => _lastResult = result ?? 'dismissed');
  }

  Future<void> _showSimpleDialog(BuildContext context) async {
    final String? result = await showDialog<String>(
      context: context,
      barrierDismissible: _barrierDismissible,
      builder: (BuildContext routeContext) => SimpleDialog(
        title: const Text('Select workspace'),
        children: <Widget>[
          SimpleDialogOption(
            onPressed: () => Navigator.pop(routeContext, 'personal'),
            child: const Text('Personal workspace'),
          ),
          SimpleDialogOption(
            onPressed: () => Navigator.pop(routeContext, 'team'),
            child: const Text('Team workspace'),
          ),
          SimpleDialogOption(
            onPressed: () => Navigator.pop(routeContext, 'guest'),
            child: const Text('Guest workspace'),
          ),
        ],
      ),
    );
    if (mounted) setState(() => _lastResult = result ?? 'dismissed');
  }

  Future<void> _showAdaptive(BuildContext context) async {
    final bool apple = switch (Theme.of(context).platform) {
      TargetPlatform.iOS || TargetPlatform.macOS => true,
      _ => false,
    };
    final String? result = await showAdaptiveDialog<String>(
      context: context,
      barrierDismissible: _barrierDismissible,
      builder: (BuildContext routeContext) => AlertDialog.adaptive(
        title: const Text('Adaptive alert'),
        content: const Text(
          'Material on desktop platforms, Cupertino on Apple platforms.',
        ),
        actions: apple
            ? <Widget>[
                cupertino.CupertinoDialogAction(
                  onPressed: () => Navigator.pop(routeContext, 'cancel'),
                  child: const Text('Cancel'),
                ),
                cupertino.CupertinoDialogAction(
                  isDefaultAction: true,
                  onPressed: () => Navigator.pop(routeContext, 'ok'),
                  child: const Text('OK'),
                ),
              ]
            : <Widget>[
                TextButton(
                  onPressed: () => Navigator.pop(routeContext, 'cancel'),
                  child: const Text('CANCEL'),
                ),
                TextButton(
                  onPressed: () => Navigator.pop(routeContext, 'ok'),
                  child: const Text('OK'),
                ),
              ],
      ),
    );
    if (mounted) setState(() => _lastResult = result ?? 'dismissed');
  }
}

File: src/Sample/Plumix.Sample/Demos/Material/DialogDemoPage.cs
using System;
using Avalonia;
using Avalonia.Media;
using Plumix.Material;
using Plumix.Rendering;
using Plumix.UI;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source: dart_sample/lib/demos/material/dialog_demo_page.dart

public sealed class DialogDemoPage : StatefulWidget
{
    public override State CreateState() => new DialogDemoPageState();

    private sealed class DialogDemoPageState : State
    {
        private bool _scrollable;
        private bool _barrierDismissible = true;
        private bool _useThemeOverrides;
        private string _lastResult = "none";

        public override Widget Build(BuildContext context)
        {
            var dialogTheme = _useThemeOverrides
                ? new DialogThemeData(
                    BackgroundColor: Color.Parse("#FFE0F2F1"),
                    IconColor: Color.Parse("#FF00695C"),
                    Shape: ShapeBorder.RoundedRectangle(14),
                    TitleTextStyle: Theme.Of(context).TextTheme.HeadlineSmall.CopyWith(color: Color.Parse("#FF004D40")),
                    BarrierColor: Color.FromArgb(0x99, 0x00, 0x4D, 0x40))
                : new DialogThemeData();
            return new DialogTheme(
                dialogTheme,
                new Builder(innerContext => BuildContent(innerContext)));
        }

        private Widget BuildContent(BuildContext context)
        {
            return new Column(
                crossAxisAlignment: CrossAxisAlignment.Stretch,
                spacing: 14,
                children:
                [
                    new Text("Dialog family", fontSize: 20),
                    new Text(
                        "Dialog, AlertDialog, SimpleDialog, typed results, intrinsic width, actions overflow, and scrollable choices.",
                        fontSize: 14,
                        color: Colors.DimGray),
                    new Row(
                        spacing: 8,
                        children:
                        [
                            ControlButton(_scrollable ? "Scrollable" : "Static", () => SetState(() => _scrollable = !_scrollable)),
                            ControlButton(_barrierDismissible ? "Barrier closes" : "Barrier locked", () => SetState(() => _barrierDismissible = !_barrierDismissible)),
                            ControlButton(_useThemeOverrides ? "Theme on" : "Theme off", () => SetState(() => _useThemeOverrides = !_useThemeOverrides)),
                        ]),
                    new Row(
                        spacing: 8,
                        children:
                        [
                            new ElevatedButton(new Text("SHOW ALERT"), () => ShowAlert(context)),
                            new OutlinedButton(new Text("SHOW DIALOG"), () => ShowPlainDialog(context)),
                            new FilledButton(new Text("SHOW SIMPLE"), () => ShowSimpleDialog(context)),
                        ]),
                    new Text($"Last result: {_lastResult}", fontSize: 13),
                ]);
        }

        private async void ShowAlert(BuildContext context)
        {
            string? result = await MaterialDialogs.ShowDialog<string>(
                context,
                routeContext => new AlertDialog(
                    icon: new Icon(Icons.InfoOutline),
                    title: new Text("Delete draft?"),
                    content: _scrollable
                        ? new Column(children:
                        [
                            new Text("This dialog keeps actions visible while the message scrolls."),
                            new SizedBox(height: 180),
                            new Text("End of the scrollable content."),
                        ])
                        : new Text("The draft can be restored later from history."),
                    scrollable: _scrollable,
                    actions:
                    [
                        new TextButton(new Text("CANCEL"), () => Navigator.Pop(routeContext, "cancel")),
                        new TextButton(new Text("DELETE"), () => Navigator.Pop(routeContext, "delete")),
                    ]),
                barrierDismissible: _barrierDismissible);
            if (Mounted) SetState(() => _lastResult = result ?? "dismissed");
        }

        private async void ShowPlainDialog(BuildContext context)
        {
            string? result = await MaterialDialogs.ShowDialog<string>(
                context,
                routeContext => new Dialog(
                    child: new Padding(
                        new Thickness(24),
                        new Column(
                            mainAxisSize: MainAxisSize.Min,
                            spacing: 16,
                            children:
                            [
                                new Text("Base Dialog", fontSize: 20),
                                new Text("This uses the same themed Material surface and route barrier."),
                                new TextButton(new Text("CLOSE"), () => Navigator.Pop(routeContext, "closed")),
                            ]))),
                barrierDismissible: _barrierDismissible);
            if (Mounted) SetState(() => _lastResult = result ?? "dismissed");
        }

        private async void ShowSimpleDialog(BuildContext context)
        {
            string? result = await MaterialDialogs.ShowDialog<string>(
                context,
                routeContext => new SimpleDialog(
                    title: new Text("Select workspace"),
                    children:
                    [
                        new SimpleDialogOption(
                            onPressed: () => Navigator.Pop(routeContext, "personal"),
                            child: new Text("Personal workspace")),
                        new SimpleDialogOption(
                            onPressed: () => Navigator.Pop(routeContext, "team"),
                            child: new Text("Team workspace")),
                        new SimpleDialogOption(
                            onPressed: () => Navigator.Pop(routeContext, "guest"),
                            child: new Text("Guest workspace")),
                    ]),
                barrierDismissible: _barrierDismissible);
            if (Mounted) SetState(() => _lastResult = result ?? "dismissed");
        }

        private static Widget ControlButton(string label, Action onPressed) =>
            new TextButton(new Text(label, fontSize: 12), onPressed);
    }
}
